### ISSUE TITLE
Interest in hatch plugin

### ISSUE THREAD

**Issue by MementoRC** [scikit-build/scikit-build-core#683]
Hello, I perused the hatchling branch and am interested in testing it on @ofek project `coincurve` that we are tidying up a bit. Currently, we have a PR that is using `scikit-build-core` successfully, but we (I?) would like to try using hatchling instead.
I had started a private plugin repository where I can evaluate how well it fits in coincurve

**Comment by henryiii** [comment]
Would https://github.com/scikit-build/scikit-build-core/pull/637 be of interest? It’s planned for the next release. Things are a bit slow since I’m on paternity leave and have less time atm.

**Comment by henryiii** [comment]
Any feedback would be extreamly useful! Especially after I review the recent changes.

**Comment by MementoRC** [comment]
Yes it is, I have been looking at this branch. Initially I did not understand why you prefer the plugin to be part of scikit-build-core, do you plan to make it an independent plugin eventually?

**Comment by MementoRC** [comment]
> Any feedback would be extreamly useful! Especially after I review the recent changes.

Certainly, I am going the route of fork/local install since it looks finished

**Comment by henryiii** [comment]
It's easier to develop, as we can still make changes to the API as needed. Yes, I hope to pull it out into a separate plugin (also setuptools) soon. In fact, might be worth making a dummy package that just pulls from scikit-build so that user code doesn't have to change the requirements when that happens.

**Comment by MementoRC** [comment]
I have invited you to a repo where I am trying out the plugin (I was not able to figure out how to use it from an installed version of the hatchling branch). For easy of use, I also changed the way to feed the info to the SettingsReader so that I can use v0.8.2

It seems to be working the way I would like it to. I would like to add a list of targets to select/verify
This would be for the case where a bigger project is used with its own scikit definitions and the hook only requests a few targets, for example just the dynamic library, or just the headers, does scikit have a cmake function for that?
